src/widgets/heightandweight/heightandweight-detailed-summary.component.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from "react";
+import { match, useParams } from "react-router";
+import SummaryCard from "../cards/summary-card.component";
+import styles from "./heightandweight-detailed-summary.css";
+import dayjs from "dayjs";
+import { getDimenionsObservationsRestAPI } from "./heightandweight.resource";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import { isEmpty } from "lodash-es";
+import {
+  convertToPounds,
+  convertToFeet,
+  convertoToInches
+} from "./heightandweight-helper";
+
+export function HeightAndWeightDetailedSummary(
+  props: HeightAndWeightDetailedSummaryProps
+) {
+  const [dimensions, setDimensions] = useState<any>({});
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+  let params = useParams();
+
+  useEffect(() => {
+    getDimenionsObservationsRestAPI(patientUuid).subscribe(response => {
+      setDimensions(
+        response.find(
+          dimension => dimension.obsData.weight.uuid === params["uuid"]
+        )
+      );
+    });
+  }, [params, patientUuid]);
+
+  function displayNoHeightAndWeight() {
+    return (
+      <SummaryCard
+        name="Height & Weight"
+        match={props.match}
+        styles={{ width: "100%" }}
+      >
+        <div className={styles.heightAndWeightDetailedSummary}>
+          <p className="omrs-bold" data-testid="errorMessage">
+            The patient's Height and Weight is not documented.
+          </p>
+          <p className="omrs-bold" data-testid="navigateLink">
+            Please <a href="/">add patient height and weight</a>.
+          </p>
+        </div>
+      </SummaryCard>
+    );
+  }
+
+  function displayHeightAndWeight() {
+    return (
+      <div className={styles.heightAndWeightDetailedSummary}>
+        <SummaryCard
+          name="Height & Weight"
+          match={props.match}
+          editBtnUrl={`/patient/dimensions`}
+          styles={{ width: "100%" }}
+        >
+          <div className={styles.heightAndWeightContainer}>
+            {!isEmpty(dimensions) && (
+              <table className={styles.summaryTable}>
+                <tbody>
+                  <tr>
+                    <td>Measured at </td>
+                    <td>
+                      {dayjs(
+                        dimensions.obsData.weight.encounter.encounterDatetime
+                      ).format("DD-MMM-YYYY")}{" "}
```

Comment:
you can create a helper for this so it can be used through out the app:
 ```function formatDate(date, format) {
          return dayjs(date).format(format);
}
```

then you can do this instead
```suggestion
                    [
                     formatDate(dimensions.obsData.weight.encounter.encounterDatetime, "DD-MMM-YYYY"),
                     formatDate(dimensions.obsData.weight.encounter.encounterDatetime, "HH:mm A")
                     ]
                     .join(" ")
```